### PR TITLE
Adding rollback handler

### PR DESCRIPTION
**Pull request by weskamm**
Make sure you have set the `MATTERMOST_HOOK_URL` in your env file before testing this!

This adds a rollback handler which deletes datastores in geoserver and files that have been downloaded when a job fails

@JakobMiksch @hblitza please review

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -58,6 +58,24 @@ services:
       - WORKERQUEUE=DeadLetterQueue
       - DEV_MODE=0
 
+  rollback-handler:
+    build:
+      context: worker/rollback-handler
+    restart: unless-stopped
+    volumes:
+      - ./geoserver_data:/home/data:Z
```
Comment (JakobMiksch) [inline docker-compose.yml:66]: currently we set the volumes all to the same directory

```suggestion
      - ./geoserver_data:/opt/geoserver_data/:Z
```

**Review by JakobMiksch — COMMENTED** [review]
Looks good so far, some remarks inline.
I need to test locally before approving.

**Comment by JakobMiksch** [comment]
This PR is targeted to the workflow of publishing a simple GeoTiff, but currently we have a new workflow that add granules to a mosaic store. We might need to adapt this rollback handler for this.

**Comment by hblitza** [comment]
Thanks for this! Tested locally - will merge now. As discussed, we will add the rollback for the new workflow later on.

**Pull request merged**